// @clerk/shared/dist/runtime/constants-BUzkV8jh.js

//#region src/internal/clerk-js/constants.ts
const PRESERVED_QUERYSTRING_PARAMS = [
	"redirect_url",
	"after_sign_in_url",
	"after_sign_up_url",
	"sign_in_force_redirect_url",
	"sign_in_fallback_redirect_url",
	"sign_up_force_redirect_url",
	"sign_up_fallback_redirect_url"
];
const CLERK_MODAL_STATE = "__clerk_modal_state";
const CLERK_SYNCED = "__clerk_synced";
const CLERK_SYNCED_STATUS = {
	NeedsSync: "false",
	Completed: "true"
};
const CLERK_SUFFIXED_COOKIES = "suffixed_cookies";
const CLERK_SATELLITE_URL = "__clerk_satellite_url";
const ERROR_CODES = {
	FORM_IDENTIFIER_NOT_FOUND: "form_identifier_not_found",
	FORM_PASSWORD_INCORRECT: "form_password_incorrect",
	FORM_PASSWORD_PWNED: "form_password_pwned",
	INVALID_STRATEGY_FOR_USER: "strategy_for_user_invalid",
	NOT_ALLOWED_TO_SIGN_UP: "not_allowed_to_sign_up",
	OAUTH_ACCESS_DENIED: "oauth_access_denied",
	OAUTH_EMAIL_DOMAIN_RESERVED_BY_SAML: "oauth_email_domain_reserved_by_saml",
	NOT_ALLOWED_ACCESS: "not_allowed_access",
	SAML_USER_ATTRIBUTE_MISSING: "saml_user_attribute_missing",
	USER_LOCKED: "user_locked",
	EXTERNAL_ACCOUNT_NOT_FOUND: "external_account_not_found",
	SESSION_EXISTS: "session_exists",
	SIGN_UP_MODE_RESTRICTED: "sign_up_mode_restricted",
	SIGN_UP_MODE_RESTRICTED_WAITLIST: "sign_up_restricted_waitlist",
	ENTERPRISE_SSO_USER_ATTRIBUTE_MISSING: "enterprise_sso_user_attribute_missing",
	ENTERPRISE_SSO_EMAIL_ADDRESS_DOMAIN_MISMATCH: "enterprise_sso_email_address_domain_mismatch",
	ENTERPRISE_SSO_HOSTED_DOMAIN_MISMATCH: "enterprise_sso_hosted_domain_mismatch",
	SAML_EMAIL_ADDRESS_DOMAIN_MISMATCH: "saml_email_address_domain_mismatch",
	INVITATION_ACCOUNT_NOT_EXISTS: "invitation_account_not_exists",
	ORGANIZATION_MEMBERSHIP_QUOTA_EXCEEDED_FOR_SSO: "organization_membership_quota_exceeded_for_sso",
	CAPTCHA_INVALID: "captcha_invalid",
	FRAUD_DEVICE_BLOCKED: "device_blocked",
	FRAUD_ACTION_BLOCKED: "action_blocked",
	SIGNUP_RATE_LIMIT_EXCEEDED: "signup_rate_limit_exceeded",
	USER_BANNED: "user_banned",
	USER_DEACTIVATED: "user_deactivated"
};
const SIGN_IN_INITIAL_VALUE_KEYS = [
	"email_address",
	"phone_number",
	"username"
];
const SIGN_UP_INITIAL_VALUE_KEYS = [
	"email_address",
	"phone_number",
	"username",
	"first_name",
	"last_name"
];
const DEBOUNCE_MS = 350;
const SIGN_UP_MODES = {
	PUBLIC: "public",
	RESTRICTED: "restricted",
	WAITLIST: "waitlist"
};
const SUPPORTED_FAPI_VERSION = "2025-11-10";
const CAPTCHA_ELEMENT_ID = "clerk-captcha";
const CAPTCHA_INVISIBLE_CLASSNAME = "clerk-invisible-captcha";

//#endregion
Object.defineProperty(exports, 'CAPTCHA_ELEMENT_ID', {
  enumerable: true,
  get: function () {
    return CAPTCHA_ELEMENT_ID;
  }
});
Object.defineProperty(exports, 'CAPTCHA_INVISIBLE_CLASSNAME', {
  enumerable: true,
  get: function () {
    return CAPTCHA_INVISIBLE_CLASSNAME;
  }
});
Object.defineProperty(exports, 'CLERK_MODAL_STATE', {
  enumerable: true,
  get: function () {
    return CLERK_MODAL_STATE;
  }
});
Object.defineProperty(exports, 'CLERK_SATELLITE_URL', {
  enumerable: true,
  get: function () {
    return CLERK_SATELLITE_URL;
  }
});
Object.defineProperty(exports, 'CLERK_SUFFIXED_COOKIES', {
  enumerable: true,
  get: function () {
    return CLERK_SUFFIXED_COOKIES;
  }
});
Object.defineProperty(exports, 'CLERK_SYNCED', {
  enumerable: true,
  get: function () {
    return CLERK_SYNCED;
  }
});
Object.defineProperty(exports, 'CLERK_SYNCED_STATUS', {
  enumerable: true,
  get: function () {
    return CLERK_SYNCED_STATUS;
  }
});
Object.defineProperty(exports, 'DEBOUNCE_MS', {
  enumerable: true,
  get: function () {
    return DEBOUNCE_MS;
  }
});
Object.defineProperty(exports, 'ERROR_CODES', {
  enumerable: true,
  get: function () {
    return ERROR_CODES;
  }
});
Object.defineProperty(exports, 'PRESERVED_QUERYSTRING_PARAMS', {
  enumerable: true,
  get: function () {
    return PRESERVED_QUERYSTRING_PARAMS;
  }
});
Object.defineProperty(exports, 'SIGN_IN_INITIAL_VALUE_KEYS', {
  enumerable: true,
  get: function () {
    return SIGN_IN_INITIAL_VALUE_KEYS;
  }
});
Object.defineProperty(exports, 'SIGN_UP_INITIAL_VALUE_KEYS', {
  enumerable: true,
  get: function () {
    return SIGN_UP_INITIAL_VALUE_KEYS;
  }
});
Object.defineProperty(exports, 'SIGN_UP_MODES', {
  enumerable: true,
  get: function () {
    return SIGN_UP_MODES;
  }
});
Object.defineProperty(exports, 'SUPPORTED_FAPI_VERSION', {
  enumerable: true,
  get: function () {
    return SUPPORTED_FAPI_VERSION;
  }
});
