// @clerk/shared/dist/runtime/constants-Bta24VLk.mjs
//#region src/constants.ts
const LEGACY_DEV_INSTANCE_SUFFIXES = [
	".lcl.dev",
	".lclstage.dev",
	".lclclerk.com"
];
const CURRENT_DEV_INSTANCE_SUFFIXES = [
	".accounts.dev",
	".accountsstage.dev",
	".accounts.lclclerk.com"
];
const DEV_OR_STAGING_SUFFIXES = [
	".lcl.dev",
	".stg.dev",
	".lclstage.dev",
	".stgstage.dev",
	".dev.lclclerk.com",
	".stg.lclclerk.com",
	".accounts.lclclerk.com",
	"accountsstage.dev",
	"accounts.dev"
];
const LOCAL_ENV_SUFFIXES = [
	".lcl.dev",
	"lclstage.dev",
	".lclclerk.com",
	".accounts.lclclerk.com"
];
const STAGING_ENV_SUFFIXES = [".accountsstage.dev"];
const LOCAL_API_URL = "https://api.lclclerk.com";
const STAGING_API_URL = "https://api.clerkstage.dev";
const PROD_API_URL = "https://api.clerk.com";
const LOCAL_FAPI_URL = "https://frontend-api.lclclerk.com";
const STAGING_FAPI_URL = "https://frontend-api.clerkstage.dev";
const PROD_FAPI_URL = "https://frontend-api.clerk.dev";
const DEFAULT_PROXY_PATH = "/__clerk";
/**
* Returns the URL for a static image
* using the new img.clerk.com service
*/
function iconImageUrl(id, format = "svg") {
	return `https://img.clerk.com/static/${id}.${format}`;
}

//#endregion
export { LOCAL_API_URL as a, PROD_API_URL as c, STAGING_ENV_SUFFIXES as d, STAGING_FAPI_URL as f, LEGACY_DEV_INSTANCE_SUFFIXES as i, PROD_FAPI_URL as l, DEFAULT_PROXY_PATH as n, LOCAL_ENV_SUFFIXES as o, iconImageUrl as p, DEV_OR_STAGING_SUFFIXES as r, LOCAL_FAPI_URL as s, CURRENT_DEV_INSTANCE_SUFFIXES as t, STAGING_API_URL as u };
