// @clerk/shared/dist/runtime/error.js
const require_clerkRuntimeError = require('./clerkRuntimeError-CaS4sM5w.js');
const require_error = require('./error-BkPoOeMv.js');

exports.ClerkAPIError = require_error.ClerkAPIError;
exports.ClerkAPIResponseError = require_error.ClerkAPIResponseError;
exports.ClerkError = require_clerkRuntimeError.ClerkError;
exports.ClerkOfflineError = require_error.ClerkOfflineError;
exports.ClerkRuntimeError = require_clerkRuntimeError.ClerkRuntimeError;
exports.ClerkWebAuthnError = require_error.ClerkWebAuthnError;
exports.EmailLinkError = require_error.EmailLinkError;
exports.EmailLinkErrorCode = require_error.EmailLinkErrorCode;
exports.EmailLinkErrorCodeStatus = require_error.EmailLinkErrorCodeStatus;
exports.MissingExpiredTokenError = require_error.MissingExpiredTokenError;
exports.buildErrorThrower = require_error.buildErrorThrower;
exports.createClerkGlobalHookError = require_error.createClerkGlobalHookError;
exports.errorToJSON = require_error.errorToJSON;
exports.is429Error = require_error.is429Error;
exports.is4xxError = require_error.is4xxError;
exports.isCaptchaError = require_error.isCaptchaError;
exports.isClerkAPIError = require_error.isClerkAPIError;
exports.isClerkAPIResponseError = require_error.isClerkAPIResponseError;
exports.isClerkError = require_clerkRuntimeError.isClerkError;
exports.isClerkRuntimeError = require_clerkRuntimeError.isClerkRuntimeError;
exports.isEmailLinkError = require_error.isEmailLinkError;
exports.isKnownError = require_error.isKnownError;
exports.isMetamaskError = require_error.isMetamaskError;
exports.isNetworkError = require_error.isNetworkError;
exports.isPasswordCompromisedError = require_error.isPasswordCompromisedError;
exports.isPasswordPwnedError = require_error.isPasswordPwnedError;
exports.isReverificationCancelledError = require_error.isReverificationCancelledError;
exports.isUnauthenticatedError = require_error.isUnauthenticatedError;
exports.isUnauthorizedError = require_error.isUnauthorizedError;
exports.isUserLockedError = require_error.isUserLockedError;
exports.parseError = require_error.parseError;
exports.parseErrors = require_error.parseErrors;
