// @clerk/shared/dist/runtime/utils/index.js
const require_runtimeEnvironment = require('../runtimeEnvironment-DJJWo5AR.js');
const require_handleValueOrFn = require('../handleValueOrFn-BmJC8Cue.js');
const require_instance = require('../instance-C1LlVxIp.js');
const require_noop = require('../noop-CVg8oVJr.js');
const require_createDeferredPromise = require('../createDeferredPromise-Bq0Rf2GM.js');
const require_utils = require('../utils-Y3tB5LFT.js');

exports.allSettled = require_utils.allSettled;
exports.createDeferredPromise = require_createDeferredPromise.createDeferredPromise;
exports.fastDeepMergeAndKeep = require_utils.fastDeepMergeAndKeep;
exports.fastDeepMergeAndReplace = require_utils.fastDeepMergeAndReplace;
exports.handleValueOrFn = require_handleValueOrFn.handleValueOrFn;
exports.isDevelopmentEnvironment = require_runtimeEnvironment.isDevelopmentEnvironment;
exports.isProductionEnvironment = require_runtimeEnvironment.isProductionEnvironment;
exports.isStaging = require_instance.isStaging;
exports.isTestEnvironment = require_runtimeEnvironment.isTestEnvironment;
exports.logErrorInDevMode = require_utils.logErrorInDevMode;
exports.noop = require_noop.noop;
exports.runIfFunctionOrReturn = require_utils.runIfFunctionOrReturn;
