// @clerk/shared/dist/runtime/internal/clerk-js/constants.js
const require_constants = require('../../constants-BUzkV8jh.js');

exports.CAPTCHA_ELEMENT_ID = require_constants.CAPTCHA_ELEMENT_ID;
exports.CAPTCHA_INVISIBLE_CLASSNAME = require_constants.CAPTCHA_INVISIBLE_CLASSNAME;
exports.CLERK_MODAL_STATE = require_constants.CLERK_MODAL_STATE;
exports.CLERK_SATELLITE_URL = require_constants.CLERK_SATELLITE_URL;
exports.CLERK_SUFFIXED_COOKIES = require_constants.CLERK_SUFFIXED_COOKIES;
exports.CLERK_SYNCED = require_constants.CLERK_SYNCED;
exports.CLERK_SYNCED_STATUS = require_constants.CLERK_SYNCED_STATUS;
exports.DEBOUNCE_MS = require_constants.DEBOUNCE_MS;
exports.ERROR_CODES = require_constants.ERROR_CODES;
exports.PRESERVED_QUERYSTRING_PARAMS = require_constants.PRESERVED_QUERYSTRING_PARAMS;
exports.SIGN_IN_INITIAL_VALUE_KEYS = require_constants.SIGN_IN_INITIAL_VALUE_KEYS;
exports.SIGN_UP_INITIAL_VALUE_KEYS = require_constants.SIGN_UP_INITIAL_VALUE_KEYS;
exports.SIGN_UP_MODES = require_constants.SIGN_UP_MODES;
exports.SUPPORTED_FAPI_VERSION = require_constants.SUPPORTED_FAPI_VERSION;
