// @clerk/shared/dist/runtime/internal/clerk-js/errors.js
const require_errors = require('../../errors-FAXDzRsM.js');

exports.clerkCoreErrorContextProviderNotFound = require_errors.clerkCoreErrorContextProviderNotFound;
exports.clerkCoreErrorNoClerkSingleton = require_errors.clerkCoreErrorNoClerkSingleton;
exports.clerkErrorDevInitFailed = require_errors.clerkErrorDevInitFailed;
exports.clerkErrorInitFailed = require_errors.clerkErrorInitFailed;
exports.clerkErrorPathRouterMissingPath = require_errors.clerkErrorPathRouterMissingPath;
exports.clerkFailedToLoadThirdPartyScript = require_errors.clerkFailedToLoadThirdPartyScript;
exports.clerkInvalidFAPIResponse = require_errors.clerkInvalidFAPIResponse;
exports.clerkInvalidRoutingStrategy = require_errors.clerkInvalidRoutingStrategy;
exports.clerkInvalidSignInUrlFormat = require_errors.clerkInvalidSignInUrlFormat;
exports.clerkInvalidSignInUrlOrigin = require_errors.clerkInvalidSignInUrlOrigin;
exports.clerkInvalidStrategy = require_errors.clerkInvalidStrategy;
exports.clerkMissingDevBrowser = require_errors.clerkMissingDevBrowser;
exports.clerkMissingFapiClientInResources = require_errors.clerkMissingFapiClientInResources;
exports.clerkMissingOptionError = require_errors.clerkMissingOptionError;
exports.clerkMissingProxyUrlAndDomain = require_errors.clerkMissingProxyUrlAndDomain;
exports.clerkMissingSignInUrlAsSatellite = require_errors.clerkMissingSignInUrlAsSatellite;
exports.clerkMissingWebAuthnPublicKeyOptions = require_errors.clerkMissingWebAuthnPublicKeyOptions;
exports.clerkNetworkError = require_errors.clerkNetworkError;
exports.clerkOAuthCallbackDidNotCompleteSignInSignUp = require_errors.clerkOAuthCallbackDidNotCompleteSignInSignUp;
exports.clerkRedirectUrlIsMissingScheme = require_errors.clerkRedirectUrlIsMissingScheme;
exports.clerkUIErrorDOMElementNotFound = require_errors.clerkUIErrorDOMElementNotFound;
exports.clerkUnsupportedEnvironmentWarning = require_errors.clerkUnsupportedEnvironmentWarning;
exports.clerkUnsupportedReloadMethod = require_errors.clerkUnsupportedReloadMethod;
exports.clerkVerifyEmailAddressCalledBeforeCreate = require_errors.clerkVerifyEmailAddressCalledBeforeCreate;
exports.clerkVerifyPasskeyCalledBeforeCreate = require_errors.clerkVerifyPasskeyCalledBeforeCreate;
exports.clerkVerifyWeb3WalletCalledBeforeCreate = require_errors.clerkVerifyWeb3WalletCalledBeforeCreate;
