// @clerk/shared/dist/runtime/internal/clerk-js/url.js
require('../../constants-WYhx_umW.js');
require('../../isomorphicAtob-Hkpnx2p8.js');
require('../../isomorphicBtoa-_pSA92Q9.js');
require('../../keys-BI9hmfja.js');
require('../../globs-D0Kpu4g1.js');
require('../../instance-C1LlVxIp.js');
require('../../url-sAcxbiir.js');
require('../../logger-BOte4sQb.js');
require('../../underscore-BX6Ucmg5.js');
require('../../path-ci3QmWY8.js');
require('../../querystring-BFj_iDi0.js');
const require_url$1 = require('../../url-CoXV4i87.js');

exports.buildURL = require_url$1.buildURL;
exports.createAllowedRedirectOrigins = require_url$1.createAllowedRedirectOrigins;
exports.getETLDPlusOneFromFrontendApi = require_url$1.getETLDPlusOneFromFrontendApi;
exports.getSearchParameterFromHash = require_url$1.getSearchParameterFromHash;
exports.hasBannedHrefProtocol = require_url$1.hasBannedHrefProtocol;
exports.hasBannedProtocol = require_url$1.hasBannedProtocol;
exports.hasExternalAccountSignUpError = require_url$1.hasExternalAccountSignUpError;
exports.hasUrlInFragment = require_url$1.hasUrlInFragment;
exports.isAllowedRedirect = require_url$1.isAllowedRedirect;
exports.isDataUri = require_url$1.isDataUri;
exports.isDevAccountPortalOrigin = require_url$1.isDevAccountPortalOrigin;
exports.isDevOrStagingUrl = require_url$1.isDevOrStagingUrl;
exports.isProblematicUrl = require_url$1.isProblematicUrl;
exports.isRedirectForFAPIInitiatedFlow = require_url$1.isRedirectForFAPIInitiatedFlow;
exports.isValidUrl = require_url$1.isValidUrl;
exports.mergeFragmentIntoUrl = require_url$1.mergeFragmentIntoUrl;
exports.pathFromFullPath = require_url$1.pathFromFullPath;
exports.relativeToAbsoluteUrl = require_url$1.relativeToAbsoluteUrl;
exports.requiresUserInput = require_url$1.requiresUserInput;
exports.sanitizeHref = require_url$1.sanitizeHref;
exports.stripOrigin = require_url$1.stripOrigin;
exports.toURL = require_url$1.toURL;
exports.trimLeadingSlash = require_url$1.trimLeadingSlash;
exports.trimTrailingSlash = require_url$1.trimTrailingSlash;
